frontend/src/metabase/core/components/TabButton/TabButton.tsx

```diff
@@ -90,15 +88,11 @@ const TabButton = forwardRef(function TabButton(
   const handleInputKeyPress: KeyboardEventHandler<HTMLInputElement> =
     useCallback(
       event => {
-        if (event.key !== "Enter") {
-          return;
-        }
-        if (typeof inputRef === "object") {
+        if (event.key === "Enter" && typeof inputRef === "object") {
           inputRef?.current?.blur();
         }
-        onFinishEditing?.();
```

Comment:
We don't actually want to call `onFinishEditing` here because then it is called twice. We already call `.blur()` on the inputRef, which will trigger the `onBlur` handler for `TabButtonInput`, which calls `onFinishEditing` already.